roles/splunk_common/tasks/get_facts.yml

```diff
@@ -123,3 +123,26 @@
     # standalone instances UNLESS they are specified by forward-servers
     - "'splunk_search_head' not in groups"
 
+- name: "Retrieve PID 1 process information (Linux)"
+  command: "ps 1"
+  register: pid1
+  changed_when: false
+  when:  ansible_system is match("Linux")
+
+- name: "Set systemd boolean"
+  set_fact:
+    splunk_systemd: "{{ pid1.stdout.find('systemd') != -1 }}"
+  when:
+    - ansible_system is match("Linux")
+
+# usually we can auto detect what the service name should be, but sometimes
+# we want to manually specify it. This is where we detect it was manually set.
+- name: "Setting service_name fact from config"
+  set_fact:
+    splunk_service_name: "{{ splunk.service_name }}"
```

Comment:
My tests in a non-dockerized/systemd Splunk installation failed here because I don't specify `service_name` parameter.
This fact will always be set to `splunk` by default if it is not manually overridden by the user. See service_name in [splunk_defaults_linux.yml](../../blob/develop/inventory/splunk_defaults_linux.yml#L76). In order to support this parameter override, maybe we should consider removing it from [splunk_defaults_linux.yml](../../blob/develop/inventory/splunk_defaults_linux.yml).
@nwang92 , @bb03?

Reply:
service_name also needs to be remove from `environ.py` to make this to work.

Reply:
It shouldn't have failed there. It should have moved on to the next task which should automatically set the service name based on if the system is running systemd and/or using the universal forwarder. Can you send me the error?

Reply:
cc @rcabral-splunk